generators/client/templates/react/package.json.ejs

```diff
@@ -195,7 +195,7 @@ limitations under the License.
     "postinstall": "webdriver-manager update --gecko false",
     <%_ } _%>
     "start": "<%= clientPackageManager %> run webpack:dev",
-    "test": "<%= clientPackageManager %> run lint && jest --coverage --logHeapUsage -w=2 --config src/test/javascript/jest.conf.js",
+    "test": "<%= clientPackageManager %> run lint && jest --coverage --updateSnapshot --logHeapUsage -w=2 --config src/test/javascript/jest.conf.js",
```

Comment:
Per my understanding, it's not recommended to update snapshots on each tests execution as it defies the purpose of snapshot testing. Instead, tests should fail in case of breaking change and snapshots should be updated/synced manually as part of fix

Reply:
so we have a different behaviour between local and cicd ? In that case, the cicd sub gen should fail for React as it doesn't contain `-u`: https://github.com/jhipster/generator-jhipster/blob/master/generators/ci-cd/templates/travis.yml.ejs#L79-L81

I'm working on switching back to NPM by default instead of Yarn, and `npm test -u` doesn't work.

Reply:
IMO, Local and CI/CD behavior should be in sync and both shouldn't contain `-u` flag to auto update snapshots. In case of breaking change, this would cause local snapshot tests to fail and would allow developer to check if this is correct failure. If yes, then, they will update snapshots by manually executing tests with -u flag. If not, they will fix UI component to preserve old behavior. 

From Jest docs:
>The test will fail if the two images do not match: either the change is unexpected, or the screenshot needs to be updated to the new version of the UI component.

Reply:
You're probably right, but I don't know how we can fix this:

```
Snapshot Summary
 › 2 snapshots failed from 1 test suite. Inspect your code changes or run `yarn test -u` to update them.
Test Suites: 1 failed, 52 passed, 53 total
Tests:       2 failed, 674 passed, 676 total
Snapshots:   2 failed, 2 total
Time:        28.924s
Ran all test suites.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
``` 

A sample repo: https://github.com/pascalgrimaud/jhipster180901-react
Travis build: https://travis-ci.com/pascalgrimaud/jhipster180901-react

Reply:
For npm foes npm test -- -u work?

Reply:
Also the reason I added -u here is that unlike normal apps our travis test generates an app and does the tests so technically there is no snapshot to compare against and in ci env jest will just fail if the snapshot test is run for first time without the -u, that's why having -u in our travis test is ok, but we shouldn't do that directly in the package.json

Reply:
@deepu105 , shouldn't snapshots required for auto generated tests be part of generator so that you don't require explicit `-u` flag for newly generated application tests to succeed?